### ISSUE TITLE
React.lazy support 

### ISSUE THREAD

**Issue by PetrSnobelt** [vercel/next.js#16516]
# Feature request

## Is your feature request related to a problem? Please describe.
We are building a UI library, which is used inside multiple projects - inside next.js app but also without next (like in CRA) and in another project using custom SSR and loadable

I'd like to use lazy loading of some sub-components (like language picker inside Navbar) - normally in next I would do it using next/dynamic, but it is not possible in this library, so I'd like to use React.lazy. But then it looks like code is not lazy-loaded (I can't see any request in network panel)

## Describe the solution you'd like
It would be great if React.lazy could be supported out of the box.
If it is not possible it would be nice to add some info about why into the documentation.

Any other advice on how to set up lazy loading inside a library which is used in multiple projects (and not all use next) could be helpful.

**Comment by fabiosantoscode** [comment]
I've been using lazy() without issues. To get the code to prefetch itself (instead of only loading ondemand) I used a webpack magic comment:

```javascript
lazy(() => import(/* webpackPrefetch: true */ './MyModule'))
```

**Comment by timneutkens** [comment]
React.lazy only works client-side. In general it's much better to use `next/dynamic` though given it will pre-render and preload the script tag needed for hydration.

**Comment by PetrSnobelt** [comment]
> React.lazy only works client-side.  

it is not a problem for us, expected usage is for something like content of popover/dropdown (only after user interaction)

> In general it's much better to use next/dynamic

I cannot use next/dynamic in CRA for example

@fabiosantoscode You are right it works nice if it is inside next app, but it doesn't work for me when it is in another npm module - probably I need to adjust webpack or babel config

**Comment by fabiosantoscode** [comment]
@PetrSnobelt Terser might be decimating the comments before they get to you, if this npm module minifies its code. Or webpack is turning it into an actual require.

If Terser is the problem, open an issue and I'll get to it eventually.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
